Automattic/wp-calypso
# Patterns: integrate Context and add second search field

Related to:
1) https://github.com/Automattic/dotcom-forge/issues/6317
2) https://github.com/Automattic/dotcom-forge/issues/6229

## Proposed Changes
With this PR we are adding the second "patterns search" field to the end of categoryPillNavigation according to the design (BnTfb6hzKcRTb2LBtuPwKe-fi-771%3A14617) and requirements (https://github.com/orgs/Automattic/projects/865/views/1?pane=issue&itemId=57872909).
Also, it was quite hard to synchronize vales between inputs and make them work together in a simple way. After thinking about it, I realized that it's good example of this my concern (https://github.com/Automattic/dotcom-forge/issues/6229#issue-2202954920) and good opportunity to tackle it together with the second search field. As a result we removed a lot of code and logic, and the flow of data is much more simple and straightforward.

## Testing Instructions
1) Open `http://calypso.localhost:3000/patterns/footer?grid=1&s=CTA`
2) Assert that search field in the header is prefilled
3) Scroll down to have sticky filters panel
4) Assert that search there is also prefilled and the field is expanded
5) Try to search any patterns, e.g. "aaa", "<empty string>" and assert that the search works well and both fields are synchronized
6) Click on X button on a field and assert that search is reset
7) Try to search something and change category (clicking on pill on navigation)
8) Assert that search is reset (search fields also empty) and you see patterns from the selected category
9) The last thing is to test responsive design
10) Also, since we injected Context, you can try to play with other properties, e.g. "Grid / list view", "Layouts / Patterns"
